### PR TITLE
Update markdown files for preprocessing purposes

### PR DESCRIPTION
**Pull request by BarkingBad**

### CODE REVIEW

**Review by julienrf — COMMENTED** [review]
Could you please check that the content of the files here is consistent with what we have on docs.scala-lang? I see some differences in the file `other-new-features/export.md`, for instance.

File: docs/docs/reference/changed-features/compiler-plugins.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ---
-layout: doc-page
 title: "Changes in Compiler Plugins"
-movedTo: https://docs.scala-lang.org/scala3/reference/changed-features/compiler-plugins.html
```
Comment (bishabosha) [inline docs/docs/reference/changed-features/compiler-plugins.md:4]: don't we need to keep the `movedTo`?

Comment (julienrf) [reply docs/docs/reference/changed-features/compiler-plugins.md:4]: No, the idea is that we will stop updating the Dotty website.

Maybe just before doing that we should remove the link “Edit this page on GitHub” from the layout, as suggested by @unkarjedy

Comment (bishabosha) [reply docs/docs/reference/changed-features/compiler-plugins.md:4]: I opened this https://github.com/lampepfl/dotty/issues/13355

**Comment by BarkingBad** [comment]
@julienrf I think the difference is because of #13327, I will update it asap, but currently I'm on vacations so if you'd like to merge it earlier you can take over this PR

File: project/Build.scala [hunk 2]
```diff
@@ -1398,7 +1398,7 @@ object Build {
               "-versions-dictionary-url",
               "https://scala-lang.org/api/versions.json",
               "-Ydocument-synthetic-types"
-            ) ++ (if (justAPI) Nil else Seq("-siteroot", "docs", "-Yapi-subdirectory")))
+            ) ++ (if (justAPI) Nil else Seq("-Yapi-subdirectory")))
```
[the remark this answers is not in the document]
Comment (BarkingBad) [reply project/Build.scala:1401]: Will check

**Comment by bishabosha** [comment]
also needs updates integrated from https://github.com/lampepfl/dotty/pull/13328 and https://github.com/lampepfl/dotty/pull/13338

File: project/Build.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (BarkingBad) [reply project/Build.scala]: I removed this if statement but left the argument `--justAPI` for convenience to generate docs without additional files like CNAME etc

**Comment by BarkingBad** [comment]
@bishabosha @julienrf  I updated the docs, I hope now everything is up to date (I rebased on top of the current master and checked if everything is applied correctly) The faster we merge it the better, since we will avoid diverging again

**Review by julienrf — COMMENTED** [review]
If I understand correctly, this PR does two things:

1. It overwrites the content of the .md documentation files of the dotty repository with the content of the repo docs.scala-lang,
2. It changes the `generateDocumentation` task such that it only renders the API documentation (the blog and static pages are not anymore part of the output).

An important consequence is that we won’t be able to generate the dotty.epfl.ch website anymore after this PR is merged. I think this is fine, but before we do that I suggest we do a last update to clearly indicate on the website that its content is frozen/archived/unmaintained, with pointers on the place to find up to date information (on every page).

Another (annoying) thing is that we must be very careful in the way we overwrite the content of the md files. We need to make sure we don’t loose any important changes that have been applied in the dotty repository (but not in docs.scala-lang). There seems to be such changes, related to experimental features…

**Comment by bishabosha** [comment]
there have been a few updates to Scala 3 docs on docs.scala-lang.org since this was last pushed

**Comment by julienrf** [comment]
@bishabosha This is not so much a big deal, actually (I mean, this is not blocking this PR). We can pick these changes later.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -501,6 +501,7 @@ jobs:
           publish_dir: docs/_site
           external_repository: lampepfl/dotty-website
           publish_branch: gh-pages
+          keep_files: true
```
Comment (BarkingBad) [inline .github/workflows/ci.yaml:504]: I believe this line will preserve older static site files while publishing nightlies according to:
https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-keeping-existing-files-keep_files

**Comment by BarkingBad** [comment]
I will add manually to `dotty-website` pregenerated static site with banners that the files are frozen since this commit.

**Comment by BarkingBad** [comment]
PoC banners for files that will be moved to docs.scala-lang eventually:
obraz

Banner for all the other static site docs:
obraz

**Comment by BarkingBad** [comment]
This PR https://github.com/lampepfl/dotty-website/pull/6 should be merged right after this one. If we merge it earlier it can be overridden once nightly docs will be published.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (BarkingBad) [reply .github/workflows/ci.yaml]: Actually this could not work (would overwrite top level index.html), so I updated `genDocs` script

**Comment by julienrf** [comment]
> This PR [lampepfl/dotty-website#6](https://github.com/lampepfl/dotty-website/pull/6) should be merged right after this one. If we merge it earlier it can be overridden once nightly docs will be published.

Should we consider publishing the nightly docs somewhere else than on dotty.epfl.ch? Or just make sure that they only override the API documentation, but nothing else.

I think lampepfl/dotty-website#6 should be merged _before_ this PR, since this PR disables the website.

**Comment by BarkingBad** [comment]
If we merged it, then the very first nightly of dotty would override the files with banners, thus we will lose the changes. We can create small PR with changes to the `genDocs` script and `Build.scala` so the files won't be overridden anymore, but we have to do it on the same day.

**Comment by BarkingBad** [comment]
> Should we consider publishing the nightly docs somewhere else than on dotty.epfl.ch? Or just make sure that they only override the API documentation, but nothing else.

I think nightlies can be still published on `dotty.epfl.ch`, since rather there is a little traffic, most people will hit the stable docs (3.x, 3.0.2). Nonetheless, the domain is not so important in that case.

**Comment by BarkingBad** [comment]
Closing since we merged #13587

**Pull request closed without merging**